### PR TITLE
Migrated ESH Add-ons

### PR DESCRIPTION
**Pull request by kaikreuzer**
What has been done in this PR:
- added all bundles from the ESH `extensions` folder
- refactored all ESH bundles to `org.openhab` namespace (artifacts as well as Java packages)
- added the Homebuilder UI from openhab-core
- merged the fragments of Classic/Basic/Paper UI from openhab-core into the UI bundles
- added Karaf features for all new add-ons

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Does anyone happen to know, what this SAT error wants to tell me?
```
[ERROR] .ui.paper/META-INF/MANIFEST.MF:[0]
The package  should be marked as "internal" if it is not exported.
```

**Comment by wborn** [comment]
It used to say that if you're not exporting a package in the manifest and "internal" is missing from the package name.

I.e. export `org.openhab.binding.somename.mypackage...`  or change it to `org.openhab.binding.somename.internal.mypackage...`

**Comment by davidgraeff** [comment]
Do the mqtt tests not work?

Regarding the UIs: I thought they go into separate repositories as they are replaced anyway within this year?

**Comment by kaikreuzer** [comment]
@davidgraeff Yes, I have issues with one MQTT test:
```
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.065 sec <<< FAILURE! - in org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest
learningMode(org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest)  Time elapsed: 0.049 sec
getDigestDataFor(org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest)  Time elapsed: 0.001 sec  <<< ERROR!
java.security.cert.CertificateException: Missing input stream
	at org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest.getDigestDataFor(PinningSSLContextProviderTest.java:50)

checkMode(org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest)  Time elapsed: 0.001 sec
pubKeyPinCallsX509CertificateGetPublicKey(org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest)  Time elapsed: 0.012 sec
certPinCallsX509CertificateGetEncoded(org.openhab.binding.mqtt.internal.ssl.PinningSSLContextProviderTest)  Time elapsed: 0 sec
Running org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest
13:04:43.445 [main] INFO  o.e.s.i.t.mqtt.MqttBrokerConnection - Starting MQTT broker connection to '10.10.0.10' with clientid BrokerHandlerTest and file store 'userdata/mqtt/10.10.0.10'
13:04:43.457 [main] INFO  o.e.s.i.t.mqtt.MqttBrokerConnection - Starting MQTT broker connection to '10.10.0.10' with clientid BrokerHandlerTest and file store 'userdata/mqtt/10.10.0.10'
13:04:43.464 [main] INFO  o.e.s.i.t.mqtt.MqttBrokerConnection - Starting MQTT broker connection to '10.10.0.10' with clientid BrokerHandlerTest and file store 'userdata/mqtt/10.10.0.10'
13:04:43.470 [main] INFO  o.e.s.i.t.mqtt.MqttBrokerConnection - Starting MQTT broker connection to '10.10.0.10' with clientid BrokerHandlerTest and file store 'userdata/mqtt/10.10.0.10'
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.041 sec - in org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest
topicVanished(org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest)  Time elapsed: 0.022 sec
firstSubscribeThenHandler(org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest)  Time elapsed: 0.006 sec
firstHandlerThanSubscribe(org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest)  Time elapsed: 0.005 sec
handlerInitializeAfterSubscribe(org.openhab.binding.mqtt.internal.MQTTTopicDiscoveryServiceTest)  Time elapsed: 0.006 sec
Running org.openhab.binding.mqtt.internal.discovery.ServiceDiscoveryServiceTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 sec - in org.openhab.binding.mqtt.internal.discovery.ServiceDiscoveryServiceTest
testDiscovery(org.openhab.binding.mqtt.internal.discovery.ServiceDiscoveryServiceTest)  Time elapsed: 0.019 sec

Results :

Tests in error: 
  PinningSSLContextProviderTest.getDigestDataFor:50 » Certificate Missing input ...

Tests run: 16, Failures: 0, Errors: 1, Skipped: 0
```
If you have a fix for that, it would be very welcome!

> Regarding the UIs: I thought they go into separate repositories 

We are talking about openHAB 2 repo here and the UIs are normal add-ons, so it is their natural place to be.

> as they are replaced anyway within this year?

They won't be replaced for openHAB 2 and what will happen for openHAB 3 has not been decided yet and should be out of scope here.

**Comment by wborn** [comment]
> The package name in that message is an empty string and the only package that exists in the bundle is one with "internal" in its name...

I found a [community thread](https://community.openhab.org/t/warnings-and-errors-on-mvn-build-syntax-validation/53943?u=wborn) with someone having the same issue. After checking the [source code](https://github.com/markus7017/org.openhab.binding.entertaintv/tree/master/src/main/java) I found that it was because there were other non-Java files (like .DS_Store) in those packages. After removing them SAT was also happy.

**Comment by kaikreuzer** [comment]
Hm, I think I solved it - I deleted all npm temp & cache folders and it started to work again. It seems the Maven build does not do a proper cleanup of the npm parts before starting.

**Comment by wborn** [comment]
I've created a PR for fixing the MQTT tests https://github.com/kaikreuzer/openhab2-addons/pull/1.

**Comment by wborn** [comment]
And I've created a PR for fixing the Jetty based tests https://github.com/kaikreuzer/openhab2-addons/pull/2.

**Comment by kaikreuzer** [comment]
Ok, @openhab/distro-maintainers, I think I am ready now - https://github.com/openhab/openhab-core/pull/469 should be merged first, then this PR and afterwards https://github.com/openhab/openhab-distro/pull/848.

I have just release 2.5.0.M1, so we are free to break everything now 😉.

**Comment by kaikreuzer** [comment]
FTR: The build now takes too long for Travis, that's why it times out after an hour.
I've disabled SAT on Travis for that reason in this PR now.
Will try to configure our Jenkins in a way that it successfully builds all PRs including SAT checks.

File: addons/binding/org.openhab.binding.astro.test/NOTICE [hunk 1]
```diff
@@ -0,0 +1,19 @@
+This content is produced and maintained by the Eclipse SmartHome project.
```
[the remark this answers is not in the document]
Comment (kaikreuzer) [reply addons/binding/org.openhab.binding.astro.test/NOTICE]: Fixed.

**Comment by kaikreuzer** [comment]
@openhab/distro-maintainers Hooray, the build is green!

**Comment by kaikreuzer** [comment]
FTR: I won't fix the sign-offs as it'll be all clean when being squashed&merged.

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ jdk: oraclejdk8
 before_install: echo "MAVEN_OPTS='-Xms1g -Xmx2g -XX:PermSize=512m -XX:MaxPermSize=1g'" > ~/.mavenrc
 
 install:
-  - echo 'mvn clean install -B -V 1> .build.stdout 2> .build.stderr' > .build.sh
+  - echo 'mvn clean install -B -DskipChecks=true -V 1> .build.stdout 2> .build.stderr' > .build.sh
```
Comment (Hilbrand) [inline .travis.yml]: This is probably needed to the missing `about.html`?
A workaround is to add the following line to the `checkstyle.properties` file:
`checkstyle.aboutHtmlCheck.url=https://raw.githubusercontent.com/openhab/openhab2-addons/master/src/etc/NOTICE`

Without the line all local builds of bindings will still give an error on the checkstyle analysis. So I think we should do something or release the fixed sat tool asap.

Comment (kaikreuzer) [reply .travis.yml]: No, it is due to https://github.com/openhab/openhab2-addons/pull/4625#issuecomment-455873630.

Comment (kaikreuzer) [reply .travis.yml]: Note that the [Jenkins PR build](https://ci.openhab.org/job/PR-openHAB2-Addons/12073/) executed the SAT checks successfully.

Comment (Hilbrand) [reply .travis.yml]: Hmm. then I have no idea why I still get that error when building locally..

**Comment by kaikreuzer** [comment]
1. I can do that in openhab-distro once the add-ons have moved.
2. This is out of scope of this PR and belongs to https://github.com/openhab/openhab-core/pull/467

**Comment by kaikreuzer** [comment]
> I added this suppression already on line 4.

Indeed, I removed the duplicate line again.

**Pull request merged**